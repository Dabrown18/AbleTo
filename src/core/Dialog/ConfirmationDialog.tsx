import React, { FunctionComponent } from 'react';
import { View } from 'react-native';
import styled from 'styled-components';

import PrimaryButton from '../Button/PrimaryButton';
import SecondaryButton from '../Button/SecondaryButton';
import TertiaryButton from '../Button/TertiaryButton';
import BodyText from '../Text/BodyText';
import HeaderText from '../Text/HeaderText';
import CustomDialog from './CustomDialog';

type Props = {
    title: string;
    body: string;
    confirmButtonMessage: string;
    declineButtonMessage?: string;
    dismissable?: boolean;
    buttonDirection?: 'column' | 'row';
    onPrimaryAction: () => void;
    onSecondaryAction?: () => void;
    onDismiss: () => void;
};

const ConfirmationDialog: FunctionComponent<Props> = ({
    title,
    body,
    confirmButtonMessage,
    declineButtonMessage,
    dismissable = true,
    buttonDirection = 'column',
    onPrimaryAction,
    onSecondaryAction,
    onDismiss,
    ...rest
}) => {
    const titleJsx = <HeaderText variant="h3">{title}</HeaderText>;
    const bodyJsx = <BodyText>{body}</BodyText>;
    const actionsJsx = (
        <StyledView buttonDirection={buttonDirection}>
            <StyledRow buttonDirection={buttonDirection}>
                <PrimaryButton title={confirmButtonMessage} wide onPress={onPrimaryAction} />
            </StyledRow>
            {declineButtonMessage ? (
                <StyledRow buttonDirection={buttonDirection}>
                    {buttonDirection === 'column' ? (
                        <StyledTertiaryButton title={declineButtonMessage} wide onPress={onSecondaryAction} />
                    ) : (
                        <StyledSecondaryButton title={declineButtonMessage} wide onPress={onSecondaryAction} />
                    )}
                </StyledRow>
            ) : null}
        </StyledView>
    );

    return (
        <CustomDialog
            titleContent={titleJsx}
            bodyContent={bodyJsx}
            actionsContent={actionsJsx}
            dismissable={dismissable}
            onDismiss={onDismiss}
            {...rest}
        />
    );
};

const StyledRow = styled(View)<{ buttonDirection: 'column' | 'row' }>`
    display: flex;
    flex: ${(props) => (props.buttonDirection === 'row' ? '1' : '0 1 auto')};
`;

const StyledView = styled(View)<{ buttonDirection: 'column' | 'row' }>`
    flex-direction: ${(props) => (props.buttonDirection === 'row' ? 'row-reverse' : 'column')};
`;

const StyledTertiaryButton = styled(TertiaryButton)`
    margin-top: 20px;
`;

const StyledSecondaryButton = styled(SecondaryButton)`
    margin-right: 10px;
`;

export default ConfirmationDialog;
