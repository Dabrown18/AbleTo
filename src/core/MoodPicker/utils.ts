import { TAU } from 'react-native-redash';

import colors from '../colors';

export const PICKER_SIZE = 220;
export const STROKE = 20;
export const KNOB_SIZE = 30;
export const BORDER_SIZE = 6;
export const R = (PICKER_SIZE - STROKE) / 2;
export const CENTER = { x: R, y: R };
const BUFFER = 0.17;

export type MoodRating = typeof MOOD_RATINGS[0];

export const normalize = (value: number) => {
    'worklet';

    const mod = value % TAU;
    return mod < 0 ? TAU + mod : mod;
};

export const MOOD_RATINGS = [
    { color: colors.teal300, valueInt: 7 },
    { color: '#B3CFBC', valueInt: 6 },
    { color: '#C8D3B3', valueInt: 5 },
    { color: colors.yellow200, valueInt: 4 },
    { color: '#F4C491', valueInt: 3 },
    { color: '#F2B487', valueInt: 2 },
    { color: colors.orange300, valueInt: 1 },
];

export const getCurrentMood = (rad: number) => {
    'worklet';

    // Adds 10 degrees (or 0.17 radians)
    // to offset the size of the rotator knob
    const radian = rad + BUFFER;

    if (radian >= 0 && radian < 0.89) {
        return MOOD_RATINGS[1];
    } else if (radian >= 0.89 && radian < 1.79) {
        return MOOD_RATINGS[0];
    } else if (radian >= 1.79 && radian < 2.69) {
        return MOOD_RATINGS[6];
    } else if (radian >= 2.69 && radian < 3.58) {
        return MOOD_RATINGS[5];
    } else if (radian >= 3.58 && radian < 4.48) {
        return MOOD_RATINGS[4];
    } else if (radian >= 4.48 && radian < 5.38) {
        return MOOD_RATINGS[3];
    } else {
        return MOOD_RATINGS[2];
    }
};

export const getRadianForCurrentMood = (moodValue: number) => {
    'worklet';
    switch (moodValue) {
        case 7:
            return 0.89 + BUFFER;
        case 6:
            return 0 + BUFFER;
        case 5:
            return 5.4 + BUFFER;
        case 4:
            return 4.48 + BUFFER;
        case 3:
            return 3.58 + BUFFER;
        case 2:
            return 2.69 + BUFFER;
        case 1:
            return 1.79 + BUFFER;
        default:
            return 0 + BUFFER;
    }
};

// Removes 1/Nth of TAU (the full circle rotation) from the current radian
// Where N is the number of mood ratings we have
export const getNextMood = (rad: number) => normalize(rad - TAU / MOOD_RATINGS.length);
