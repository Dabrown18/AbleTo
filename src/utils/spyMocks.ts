import { AccessibilityInfo } from 'react-native';

import { mockAddEventListener } from './mocks/accessibilityInfo';

// Add mocks that we'd like to spy on globally in any tests.
// This file is referenced in `jest.config.js`.
jest.spyOn(AccessibilityInfo, 'addEventListener').mockImplementation(mockAddEventListener as any);
