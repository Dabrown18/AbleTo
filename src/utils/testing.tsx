import { NavigationContainer } from '@react-navigation/native';
import { render } from '@testing-library/react-native';
import React, { ReactNode } from 'react';
import { Provider as PaperProvider } from 'react-native-paper';
import { QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';

import server from './mocks/server';
import { rootNavigation } from '@src/activities/controller/ActivityController';
import linking from '@src/config/deeplinks';
import queryClient from '@src/config/queryClient';
import colors from '@src/core/colors';
import fonts from '@src/core/fonts';
import LoadingIndicator from '@src/core/LoadingIndicator/LoadingIndicator';
import store from '@src/redux/store';

const theme = {
    colors,
    fonts,
};

// Globally exposed testing utils that can be reused across multiple files

export const renderWithTheme = (children: ReactNode) => render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);

// Currently used in tests that involve controller initialization
// as the controller uses the navigator ref
export const renderWithNavigator = (children: ReactNode) => {
    return render(<NavigationContainer ref={rootNavigation}>{children}</NavigationContainer>);
};

const applyProviders = (children: ReactNode) => {
    return (
        <Provider store={store}>
            <QueryClientProvider client={queryClient}>
                <ThemeProvider theme={theme}>
                    <NavigationContainer linking={linking} ref={rootNavigation} fallback={<LoadingIndicator />}>
                        <PaperProvider>{children}</PaperProvider>
                    </NavigationContainer>
                </ThemeProvider>
            </QueryClientProvider>
        </Provider>
    );
};

export const renderWithAppProviders = (children: ReactNode) => render(applyProviders(children));

export const createWithAppProviders = (children: ReactNode) => TestRenderer.create(applyProviders(children));

/**
 * Sets up a request interception layer.
 * Should be called when tests involve API calls.
 * @param clearCache Clear QueryClient cache. Defaults to true.
 */
export const initServer = (clearCache = true) => {
    // Sets up the requests interception layer before all tests.
    beforeAll(() => server.listen());
    // Removes any request handlers that were added on runtime
    // and resets QueryClient as consecutive requests to the
    // same endpoint get cached.
    afterEach(() => {
        server.resetHandlers();
        if (clearCache) queryClient.clear();
    });
    // Clean up after all tests are done, preventing this
    // interception layer from affecting irrelevant tests.
    afterAll(() => server.close());
};
