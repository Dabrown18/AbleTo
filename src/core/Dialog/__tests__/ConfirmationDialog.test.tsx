import { fireEvent } from '@testing-library/react-native';
import React from 'react';

import ConfirmationDialog from '../ConfirmationDialog';
import { renderWithTheme } from '@src/utils/testing';

const confirmationDialogTitle = 'Title';
const confirmationDialogBody = 'Body';
const confirmButtonMessage = 'Yes';
const declineButtonMessage = 'No';
const dismissable = false;
const onPrimaryAction = jest.fn();
const onSecondaryAction = jest.fn();
const onDismiss = jest.fn();
const dialog = (
    <ConfirmationDialog
        title={confirmationDialogTitle}
        body={confirmationDialogBody}
        confirmButtonMessage={confirmButtonMessage}
        declineButtonMessage={declineButtonMessage}
        onPrimaryAction={() => {
            onPrimaryAction();
        }}
        onSecondaryAction={() => {
            onSecondaryAction();
        }}
        onDismiss={() => {
            onDismiss();
        }}
    />
);

describe('CustomDialog Snapshots', () => {
    test('Dismissable', () => {
        const tree = renderWithTheme(dialog).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Not Dismissable', () => {
        const tree = renderWithTheme(
            <ConfirmationDialog
                title={confirmationDialogTitle}
                body={confirmationDialogBody}
                confirmButtonMessage={confirmButtonMessage}
                declineButtonMessage={declineButtonMessage}
                dismissable={dismissable}
                onPrimaryAction={() => {
                    onPrimaryAction();
                }}
                onSecondaryAction={() => {
                    onSecondaryAction();
                }}
                onDismiss={() => {
                    onDismiss();
                }}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});

describe('Confirmation Dialog Behaviour', () => {
    it('Handles Confirm Button onClick', async () => {
        const { findByText } = renderWithTheme(dialog);

        const button = await findByText(confirmButtonMessage);

        fireEvent.press(button);

        expect(onPrimaryAction).toHaveBeenCalledTimes(1);
    });

    it('Handles Decline Button onClick', async () => {
        const { findByText } = renderWithTheme(dialog);

        const button = await findByText(declineButtonMessage);

        fireEvent.press(button);

        expect(onSecondaryAction).toHaveBeenCalledTimes(1);
    });
});
