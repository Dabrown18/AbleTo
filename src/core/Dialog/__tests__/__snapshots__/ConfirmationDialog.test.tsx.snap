// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomDialog Snapshots Dismissable 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#2D292680",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
    testID="Dismissable"
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 12,
            "marginBottom": 0,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 0,
            "paddingBottom": 20,
            "paddingLeft": 18,
            "paddingRight": 18,
            "paddingTop": 20,
            "width": "auto",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "justifyContent": "center",
              "position": "relative",
            },
          ]
        }
      >
        <View
          hasCloseButton={false}
          style={
            Array [
              Object {
                "marginBottom": 8,
                "marginTop": 0,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2D2926",
                  "fontFamily": "guardian_egyp_light",
                  "fontSize": 24,
                  "fontWeight": "300",
                  "lineHeight": 34,
                },
              ]
            }
            variant="h3"
          >
            Title
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 18,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#4B4845",
                "fontFamily": "fakt_pro_normal",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
            ]
          }
          variant="regular"
        >
          Body
        </Text>
      </View>
      <View>
        <View
          buttonDirection="column"
          style={
            Array [
              Object {
                "flexDirection": "column",
              },
            ]
          }
        >
          <View
            buttonDirection="column"
            style={
              Array [
                Object {
                  "display": "flex",
                  "flexBasis": "auto",
                  "flexGrow": 0,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                accessibilityLabel="Yes"
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "borderRadius": 100,
                      "borderWidth": 1.5,
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "minWidth": "100%",
                      "paddingBottom": 12,
                      "paddingLeft": 24,
                      "paddingRight": 24,
                      "paddingTop": 12,
                    },
                    Object {
                      "backgroundColor": "#85356F",
                      "borderColor": "#85356F",
                    },
                    Object {
                      "elevation": 0,
                    },
                    undefined,
                  ]
                }
                wide={true}
              >
                <Text
                  importantForAccessibility="no-hide-descendants"
                  style={
                    Array [
                      Object {
                        "color": "#4B4845",
                        "fontFamily": "fakt_pro_medium",
                        "fontSize": 16,
                        "fontWeight": "500",
                        "lineHeight": 24,
                      },
                      Object {
                        "flexShrink": 1,
                      },
                      undefined,
                      Object {
                        "color": "#fff",
                      },
                    ]
                  }
                  variant="h4"
                >
                  Yes
                </Text>
              </View>
            </View>
          </View>
          <View
            buttonDirection="column"
            style={
              Array [
                Object {
                  "display": "flex",
                  "flexBasis": "auto",
                  "flexGrow": 0,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                accessibilityLabel="No"
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "borderRadius": 100,
                      "borderWidth": 1.5,
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "minWidth": "100%",
                      "paddingBottom": 12,
                      "paddingLeft": 24,
                      "paddingRight": 24,
                      "paddingTop": 12,
                    },
                    Object {
                      "backgroundColor": "#F3E8F000",
                      "borderColor": "transparent",
                    },
                    Array [
                      Object {
                        "marginTop": 20,
                      },
                    ],
                    Object {
                      "backgroundColor": undefined,
                    },
                  ]
                }
                wide={true}
              >
                <Text
                  importantForAccessibility="no-hide-descendants"
                  style={
                    Array [
                      Object {
                        "color": "#4B4845",
                        "fontFamily": "fakt_pro_medium",
                        "fontSize": 16,
                        "fontWeight": "500",
                        "lineHeight": 24,
                      },
                      Object {
                        "flexShrink": 1,
                      },
                      undefined,
                      Object {
                        "color": "#85356F",
                      },
                    ]
                  }
                  variant="h4"
                >
                  No
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`CustomDialog Snapshots Not Dismissable 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#2D292680",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
    testID="Not Dismissable"
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 12,
            "marginBottom": 0,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 0,
            "paddingBottom": 20,
            "paddingLeft": 18,
            "paddingRight": 18,
            "paddingTop": 20,
            "width": "auto",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "justifyContent": "center",
              "position": "relative",
            },
          ]
        }
      >
        <View
          hasCloseButton={false}
          style={
            Array [
              Object {
                "marginBottom": 8,
                "marginTop": 0,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2D2926",
                  "fontFamily": "guardian_egyp_light",
                  "fontSize": 24,
                  "fontWeight": "300",
                  "lineHeight": 34,
                },
              ]
            }
            variant="h3"
          >
            Title
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 18,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#4B4845",
                "fontFamily": "fakt_pro_normal",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
            ]
          }
          variant="regular"
        >
          Body
        </Text>
      </View>
      <View>
        <View
          buttonDirection="column"
          style={
            Array [
              Object {
                "flexDirection": "column",
              },
            ]
          }
        >
          <View
            buttonDirection="column"
            style={
              Array [
                Object {
                  "display": "flex",
                  "flexBasis": "auto",
                  "flexGrow": 0,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                accessibilityLabel="Yes"
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "borderRadius": 100,
                      "borderWidth": 1.5,
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "minWidth": "100%",
                      "paddingBottom": 12,
                      "paddingLeft": 24,
                      "paddingRight": 24,
                      "paddingTop": 12,
                    },
                    Object {
                      "backgroundColor": "#85356F",
                      "borderColor": "#85356F",
                    },
                    Object {
                      "elevation": 0,
                    },
                    undefined,
                  ]
                }
                wide={true}
              >
                <Text
                  importantForAccessibility="no-hide-descendants"
                  style={
                    Array [
                      Object {
                        "color": "#4B4845",
                        "fontFamily": "fakt_pro_medium",
                        "fontSize": 16,
                        "fontWeight": "500",
                        "lineHeight": 24,
                      },
                      Object {
                        "flexShrink": 1,
                      },
                      undefined,
                      Object {
                        "color": "#fff",
                      },
                    ]
                  }
                  variant="h4"
                >
                  Yes
                </Text>
              </View>
            </View>
          </View>
          <View
            buttonDirection="column"
            style={
              Array [
                Object {
                  "display": "flex",
                  "flexBasis": "auto",
                  "flexGrow": 0,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                accessibilityLabel="No"
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "borderRadius": 100,
                      "borderWidth": 1.5,
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "minWidth": "100%",
                      "paddingBottom": 12,
                      "paddingLeft": 24,
                      "paddingRight": 24,
                      "paddingTop": 12,
                    },
                    Object {
                      "backgroundColor": "#F3E8F000",
                      "borderColor": "transparent",
                    },
                    Array [
                      Object {
                        "marginTop": 20,
                      },
                    ],
                    Object {
                      "backgroundColor": undefined,
                    },
                  ]
                }
                wide={true}
              >
                <Text
                  importantForAccessibility="no-hide-descendants"
                  style={
                    Array [
                      Object {
                        "color": "#4B4845",
                        "fontFamily": "fakt_pro_medium",
                        "fontSize": 16,
                        "fontWeight": "500",
                        "lineHeight": 24,
                      },
                      Object {
                        "flexShrink": 1,
                      },
                      undefined,
                      Object {
                        "color": "#85356F",
                      },
                    ]
                  }
                  variant="h4"
                >
                  No
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
